Packs/BluelivThreatContext/Integrations/BluelivThreatContext/BluelivThreatContext.py

```diff
@@ -0,0 +1,1062 @@
+        res = self._http_request(method='POST', url_suffix='/gateway', json_data=body, headers=self._headers)
+        return res
+
+    def get_threat_actor_info(self, threat_actor_id):
+        url = "threat-actor/{}".format(threat_actor_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_campaign_info(self, campaign_id: str):
+        url = "campaign/{}".format(campaign_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_malware_hash_info(self, file_hash, hash_type="md5"):
+        url = "malware/?dork={}%3A%22{}%22".format(hash_type, file_hash)
+        result = self._query_gateway(url)
+        return result
+
+    def get_malware_info(self, malware_id):
+        url = "malware/{}".format(malware_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_ip_info(self, ip_id):
+        url = "ip/{}".format(ip_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_fqdn_info(self, fqdn_id):
+        url = "fqdn/{}".format(fqdn_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_crime_server_info(self, cs_id):
+        url = "crime-server/{}".format(cs_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_attack_pattern_info(self, attack_pattern_id):
+        url = "attack-pattern/{}".format(attack_pattern_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_tool_info(self, tool_id):
+        url = "tool/{}".format(tool_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_signature_info(self, signature_id):
+        url = "signature/{}".format(signature_id)
+        result = self._query_gateway(url)
+        return result
+
+    def get_cve_info(self, cve_id):
+        url = "cve/{}".format(cve_id)
+        result = self._query_gateway(url)
+        return result
+
+    def search_by_name(self, key, value):
+        value = value.replace(' ', '+')
+        if key in SEARCHABLE_BY_NAME:
+            url = "{}/?fuzzy_filter%5Bname%5D={}".format(key, value)
+        if key in SEARCHABLE_BY_HASH:
+            url = "indicator/?fuzzy_filter%5Bvalue%5D={}".format(value)
+        if key == 'crime-server':
+            url = "crime-server/?fuzzy_filter%5Bcrime_server_url%5D={}".format(value)
+        if key == 'fqdn':
+            url = "fqdn/?fuzzy_filter%5Bdomain%5D={}".format(value)
+        if key == 'ip':
+            url = "ip/?fuzzy_filter%5Baddress%5D={}".format(value)
+
+        result = self._query_gateway(url)
+        return result.get("data", [])[0].get("id", "0")
+
+    def get_relationships(self, object_name, value, of):
+        url = "{}/{}/relationships/{}/".format(object_name, value, of)
+        result = self._query_gateway(url)
+        ids = ""
+        if result != "error":
+            for item in result['data']:
+                ids = ids + str(item['id']) + ","
```

Comment:
this will result with a comma at the end of `ids`, are you sure this is valid?

better pythonic way:
```suggestion
            ids = ','.join(str(item['id']) for item in result['data'])
```